processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -107,15 +105,15 @@ public Sequence<Cursor> makeCursors(
                   retVal.getColumnSelectorFactory(),
                   dimensionToUnnest,
                   outputColumnName,
-                  allowSet
+                  filter
```

Comment:
Above, we _may_ have passed the filter into the "base". If we did, do we want to reapply the same filter here? Or, should the above code actually be:

```java
    final Filter forBaseFilter;
    final Filter localFilter;
    if (filter == null || filter.getRequiredColumns().contains(outputColumnName) ) {
      forBaseFilter = filter;
      localFilter = null;
    } else {
      forBaseFilter = filter;
      localFilter = null;
   }
```

Then pass `localFilter` where we're passing `filter`?

Reply:
The filter that appears here will be on the unnested column only. For example if you are unnesting dim3 to ud3, any filter on ud3 will appear here. The filter on ud3 cannot be pushed into base but can only be sent to the unnest cursor as an allow filter. There is scope of optimization here to rewrite the same filter on dim3 and push it to the base